### PR TITLE
Mark ourself as connected

### PR DESCRIPTION
**Pull request by StephenButtolph**
See: https://github.com/ava-labs/avalanchego/pull/2981

### CODE REVIEW

**Review by JacobEverly — APPROVED** [review]

File: peer/network.go [hunk 1]
```diff
@@ -505,12 +505,11 @@ func (n *network) Connected(ctx context.Context, nodeID ids.NodeID, nodeVersion
 		return nil
 	}
 
-	if nodeID == n.self {
+	if nodeID != n.self {
 		log.Debug("skipping registering self as peer")
-		return nil
+		n.peers.Connected(nodeID, nodeVersion)
 	}
```
Comment (lebdron) [inline peer/network.go:511]: I had the same idea initially, but having `nodeID != n.self` and logging `skipping registering self as peer` might be confusing. I ended up with this.
```go
if nodeID != n.self {
    n.peers.Connected(nodeID, nodeVersion)
} else {
    log.Debug("skipping registering self as peer")
}
```

**Comment by lebdron** [comment]
Should close https://github.com/ava-labs/coreth/issues/515

File: peer/network.go (hunk 1 above)
Comment (StephenButtolph) [reply peer/network.go:511]: I think there is some work to be done in the p2p sdk to standardize how it handles the registration of itself as a peer. So this PR is likely to change a bit.

Comment (StephenButtolph) [reply peer/network.go:511]: But yeah as this PR is currently written that log is incorrect.

File: peer/network.go [hunk 2]
```diff
@@ -505,12 +505,11 @@ func (n *network) Connected(ctx context.Context, nodeID ids.NodeID, nodeVersion
 		return nil
 	}
 
-	if nodeID == n.self {
-		log.Debug("skipping registering self as peer")
-		return nil
+	if nodeID != n.self {
+		// The legacy peer tracker doesn't expect to be connected to itself.
+		n.peers.Connected(nodeID, nodeVersion)
 	}
```
Comment (ceyonur) [inline peer/network.go:511]: I guess it's safe but should we also skip disconnecting self in legacy peer tracker as well? (or at least we can add a comment)

**Review by ceyonur — APPROVED** [review]

**Pull request merged**